src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -127,12 +127,18 @@ void regclass_Core(py::module m) {
             py::arg("device_name"),
             py::arg("config") = py::dict());
 
-    cls.def(
-        "add_extension",
-        [](ov::runtime::Core& self, const std::string& library_path) {
-            return self.add_extension(library_path);
-        },
-        py::arg("library_path"));
+    cls.def("add_extension",
+            (void (ov::runtime::Core::*)(const std::string&)) & ov::runtime::Core::add_extension,
```

Comment:
why do we use lambda in some cases, but in other we use c-cast?
 besides that c-cast is undesirable in c++, it's better to use static_cast, e.g. 
static_cast<void (ov::runtime::Core::*)(const std::string&)>(& ov::runtime::Core::add_extension)

So we have a choice between static_cast<>() or lambda. I prefer the 2nd option, it's more readable 
Let's pick one style. @akuporos what do you think?

Reply:
We use lambdas when it needs to be preprocessed on bindings side, so i thing we can use static_cast here for now. But anyway, we can update it later to use only lambdas in favor of readability, cc @jiwaszki, @akuporos